File: rust/007/main.rs
/*
    ProjectEuler 007
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    let mut count = 0;
    let mut primes = vec![];
    let mut n = 1;
    while count < 10001 {
        n += 1;
        let mut is_prime = true;
        for p in primes.iter() {
            if n % p == 0 {
                is_prime = false;
                break;
            }
        }
        if is_prime {
            primes.push(n);
            count += 1;
        }
    }
    println!("{}", n);
}

File: lua/007/main.lua
--[[
    ProjectEuler 007
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local count = 0
    local ps = {}
    local n = 2
    while count < 10001 do
        local isprime = true
        for i, p in ipairs(ps) do
            if n % p == 0 then
                isprime = false
                break
            end
        end
        if isprime then
            table.insert(ps, n)
            count = count + 1
        end
        n = n + 1
    end
    print(ps[count])
end

main()
